x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curations_logic.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+import { DEFAULT_META } from '../../../shared/constants';
+
+import {
+  clearFlashMessages,
+  setSuccessMessage,
+  flashAPIErrors,
+} from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+import { DELETE_MESSAGE, SUCCESS_MESSAGE } from './constants';
+import { Curation, CurationsAPIResponse } from './types';
+
+interface CurationsValues {
+  dataLoading: boolean;
+  curations: Curation[];
+  meta: Meta;
+}
+
+interface CurationsActions {
+  onCurationsLoad(response: CurationsAPIResponse): CurationsAPIResponse;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+  loadCurations(): void;
+  deleteCurationSet(id: string): string;
+}
+
+export const CurationsLogic = kea<MakeLogicType<CurationsValues, CurationsActions>>({
+  path: ['enterprise_search', 'app_search', 'curations_logic'],
+  actions: () => ({
+    onCurationsLoad: ({ results, meta }) => ({ results, meta }),
+    onPaginate: (newPageIndex) => ({ newPageIndex }),
+    loadCurations: true,
+    deleteCurationSet: (id) => id,
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        loadCurations: () => true,
+        onCurationsLoad: () => false,
+      },
+    ],
+    curations: [
+      [],
+      {
+        onCurationsLoad: (_, { results }) => results,
+      },
+    ],
+    meta: [
+      DEFAULT_META,
+      {
+        onCurationsLoad: (_, { meta }) => meta,
+        onPaginate: (state, { newPageIndex }) => {
+          const newState = { page: { ...state.page } };
+          newState.page.current = newPageIndex;
```

Comment:
I think you could accomplish this with `const newState = { page: { ...stage.page, current: newPageIndex} };`

Reply:
This will get refactored out in upcoming pagination PR refactor, and I _think_ I do something along those lines